File: FoodQuiz/app/src/main/java/com/example/tipper/CustomXAxisValueFormatter.java
package com.example.tipper;

import com.github.mikephil.charting.components.AxisBase;
import com.github.mikephil.charting.formatter.ValueFormatter;

public class CustomXAxisValueFormatter extends ValueFormatter {

    private final String[] labels;

    public CustomXAxisValueFormatter(String[] labels) {
        this.labels = labels;
    }

    @Override
    public String getAxisLabel(float value, AxisBase axis) {
        // Ensure the value is within range
        int index = Math.min((int) value, labels.length - 1);
        return labels[index];
    }
}

File: KotlinFoodApp/app/src/main/java/com/example/kotlinfoodapp/CustomXAxisValueFormatter.kt
package com.example.kotlinfoodapp

import com.github.mikephil.charting.components.AxisBase
import com.github.mikephil.charting.formatter.ValueFormatter

class CustomXAxisValueFormatter(private val labels: Array<String>) : ValueFormatter() {
    override fun getAxisLabel(value: Float, axis: AxisBase): String {
        // Ensure the value is within range
        val index = value.toInt().coerceAtMost(labels.size - 1)
        return labels[index]
    }
}





